src/Uno.UWP/UI/Core/Internal/KeyboardStateTracker.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using Windows.System;
+using Windows.UI.Core;
+
+namespace Uno.UI.Core.Internal
+{
+	internal static class KeyboardStateTracker
+	{
+		private static readonly Dictionary<VirtualKey, CoreVirtualKeyStates> _keyStates = new Dictionary<VirtualKey, CoreVirtualKeyStates>();
+
+		internal static CoreVirtualKeyStates GetKeyState(VirtualKey key)
+		{
+			if (_keyStates.TryGetValue(key, out var state))
+			{
+				return state;
+			}
+
+			return CoreVirtualKeyStates.None;
+		}
+
+		internal static void OnKeyDown(VirtualKey key)
+		{
+			if (!_keyStates.ContainsKey(key))
+			{
+				_keyStates[key] = CoreVirtualKeyStates.None;
+			}
+
+			if (!_keyStates[key].HasFlag(CoreVirtualKeyStates.Locked))
+			{
+				_keyStates[key] = CoreVirtualKeyStates.Down | CoreVirtualKeyStates.Locked;
+			}
+			else
+			{
+				_keyStates[key] = CoreVirtualKeyStates.Down;
+			}
+
+			SetStateOnNonSideKeys(key);
+		}
+
+		internal static void OnKeyUp(VirtualKey key)
+		{
+			if (!_keyStates.ContainsKey(key))
+			{
+				_keyStates[key] = CoreVirtualKeyStates.None;
+			}
+
+			if (_keyStates[key].HasFlag(CoreVirtualKeyStates.Locked))
+			{
+				_keyStates[key] = CoreVirtualKeyStates.None | CoreVirtualKeyStates.Locked;
```

Comment:
You have logic reacting to the `Locked` flag, but there doesn't seem to be any way for `Locked` to actually get introduced? Future-proofing?